src/tests/ftest/posix/simul.py

```diff
@@ -0,0 +1,157 @@
+from avocado.core.exceptions import TestFail
+from dfuse_test_base import DfuseTestBase
+from env_modules import load_mpi
+from general_utils import DaosTestError
+from general_utils import run_command
+
+
+class PosixSimul(DfuseTestBase):
+    # pylint: disable=too-few-public-methods
+    # pylint: disable=too-many-ancestors
+    """Tests a posix container with simul
+
+    From : https://github.com/LLNL/simul
+
+    "simul" is an MPI coordinated test of parallel filesystem system calls and
+    library functions.  It was designed to perform filesystem operations
+    simultaneously from many nodes and processes to test the correctness
+    and coherence of parallel filesystems.
+
+    List of tests:
+        Test #0: open, shared mode.
+        Test #1: close, shared mode.
+        Test #2: file stat, shared mode.
+        Test #3: lseek, shared mode.
+        Test #4: read, shared mode.
+        Test #5: write, shared mode.
+        Test #6: chdir, shared mode.
+        Test #7: directory stat, shared mode.
+        Test #8: statfs, shared mode.
+        Test #9: readdir, shared mode.
+        Test #10: mkdir, shared mode.
+        Test #11: rmdir, shared mode.
+        Test #12: unlink, shared mode.
+        Test #13: rename, shared mode.
+        Test #14: creat, shared mode.
+        Test #15: truncate, shared mode.
+        Test #16: symlink, shared mode.
+        Test #17: readlink, shared mode.
+        Test #18: link to one file, shared mode.
+        Test #19: link to a file per process, shared mode.
+        Test #20: fcntl locking, shared mode.
+        Test #21: open, individual mode.
+        Test #22: close, individual mode.
+        Test #23: file stat, individual mode.
+        Test #24: lseek, individual mode.
+        Test #25: read, individual mode.
+        Test #26: write, individual mode.
+        Test #27: chdir, individual mode.
+        Test #28: directory stat, individual mode.
+        Test #29: statfs, individual mode.
+        Test #30: readdir, individual mode.
+        Test #31: mkdir, individual mode.
+        Test #32: rmdir, individual mode.
+        Test #33: unlink, individual mode.
+        Test #34: rename, individual mode.
+        Test #35: creat, individual mode.
+        Test #36: truncate, individual mode.
+        Test #37: symlink, individual mode.
+        Test #38: readlink, individual mode.
+        Test #39: link to one file, individual mode.
+        Test #40: link to a file per process, individual mode.
+        Test #41: fcntl locking, individual mode.
+
+    :avocado: recursive
+    """
+
+    def run_simul(self, include=None, exclude=None):
+        """ Run simul
+
+        include str:  comma-separated list of tests to include
+        exclude str:  comma-separated list of tests to exclude
+
+        If include value is set, exclude value is ignored and vice versa.
+
+        """
+        # Assuming all vms use the same OS
+        host_os = distro.linux_distribution()[0].lower()
```

Comment:
FYI, avocado provides a method for obtaining the OS:
```
from avocado.utils.distro import detect
```
`detect` returns a https://github.com/avocado-framework/avocado/blob/69lts/avocado/utils/distro.py#L35 object.
```

Reply:
Why would avocado duplicate the effort to code their own distro library?